tokio-tls/Cargo.toml

```diff
@@ -35,6 +38,18 @@ cfg-if = "0.1"
 env_logger = { version = "0.6", default-features = false }
 futures = { version = "0.3.0", features = ["async-await"] }
 
+[features]
+tokio-nativetls = [
+  "native-tls"
+]
+tokio-rustls = [
+  "rustls",
+  "webpki",
+]
+default = [
```

Comment:
What if we avoid a `default` implementation and require opt-in to an implementation. (https://github.com/tokio-rs/tokio/issues/1268#issuecomment-553162845).

In this case, when neither `rustls` nor `native-tls` are selected, the crate would be an empty "shim". It would panic if used w/o picking one.

The question is what to do if *both* are enabled. In that case, I would say there should be a runtime flag to pick between them?

Reply:
Totally, so if I understand correctly, is the point is to not favor one implementation over the other? I get that logic, but I wonder if it makes sense in the short term to default to `native-tls` so that the code remains backwards compatible for the time being -- then this change could be released without a major version update.